lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -648,6 +648,14 @@ defmodule Kernel.ErrorsTest do
     assert_compile_fail CompileError,
       "nofile:1: cannot use ^x outside of match clauses",
       'x = 8; <<a, b::size(^x)>> = <<?a, ?b>>'
+
+    assert_compile_fail CompileError,
+      "nofile:2: size in bitstring expects an integer or a variable as argument, got: bar()",
+      '''
+      defmodule Kernel.ErrorsTest.InvalidSizeInBitStrings do
+        def foo(<<_::size(bar)>>), do: :ok
```

Comment:
Re-tested this and I need the warning to have a breaking test, the ambiguous variable expansion to function is necessary to cause the error